### ISSUE TITLE
GraphQL types not generated

### ISSUE THREAD

**Issue by Mosnar** [verbb/formie#388]
**Description**
This is my first time trying the GraphQL mutations for a project, so pardon me if I'm just doing something wrong.

I have a bunch of complex field types, such as groups and repeaters; however, none of them appear to be available in my Full Schema. I'm following the docs on mutations and expecting to find types following the format: `{formHandle}_{fieldHandle}_FormieGroupInput`; however, the schema is reporting these fields as "String"

For example, I created a new group field called "testgroup" with a single text field in it. The submission type generated for this form includes:
```
testgroup: String
```
when I'm expecting: `testform_testgroup_FormieGroupInput`

### Form settings
Please pick from the options below.

- Multi-page form: No
- Submission Method: N/A
- Client-side Validation: N/A
- Custom Form Templates: N/A

**Additional info**
- Plugin version: 1.3.21
- Craft version: 3.6.9
- Multi-site: No

**Comment by engram-design** [comment]
Yep, refer to https://github.com/verbb/formie/issues/299 and https://github.com/verbb/formie/issues/368 as Group and Repeater support (for querying or mutations) is underway.

However, this is scheduled for the next release, and to get this early to test out, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.3.21",
  "...": "..."
}
```

Then run `composer update`.

**Comment by Mosnar** [comment]
@engram-design Awesome, thanks! I tried out the branch but am getting a warning notice on the frontend of the site.

```
yii\base\ErrorException: Trying to get property 'handle' of non-object in /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/gql/types/input/NameInputType.php:26
Stack trace:
#0 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/ErrorHandler.php(77): yii\base\ErrorHandler->handleError(8, 'Trying to get p...', '/Users/ransom/P...', 26)
#1 /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/gql/types/input/NameInputType.php(26): craft\web\ErrorHandler->handleError(8, 'Trying to get p...', '/Users/ransom/P...', 26, Array)
#2 /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/fields/formfields/Name.php(524): verbb\formie\gql\types\input\NameInputType::getType(Object(verbb\formie\fields\formfields\Name))
#3 /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/gql/types/input/GroupInputType.php(32): verbb\formie\fields\formfields\Name->getContentGqlMutationArgumentType()
#4 /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/fields/formfields/Group.php(193): verbb\formie\gql\types\input\GroupInputType::getType(Object(verbb\formie\fields\formfields\Group))
#5 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/gql/base/Mutation.php(39): verbb\formie\fields\formfields\Group->getContentGqlMutationArgumentType()
#6 /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/gql/mutations/SubmissionMutation.php(90): craft\gql\base\Mutation::prepareResolver(Object(verbb\formie\gql\resolvers\mutations\SubmissionResolver), Array)
#7 /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/gql/mutations/SubmissionMutation.php(44): verbb\formie\gql\mutations\SubmissionMutation::createSaveMutation(Object(verbb\formie\elements\Form))
#8 /Users/ransom/Projects/Work/<redacted>/vendor/verbb/formie/src/Formie.php(317): verbb\formie\gql\mutations\SubmissionMutation::getMutations()
#9 [internal function]: verbb\formie\Formie->verbb\formie\{closure}(Object(craft\events\RegisterGqlMutationsEvent))
#10 /Users/ransom/Projects/Work/<redacted>/vendor/yiisoft/yii2/base/Event.php(312): call_user_func(Object(Closure), Object(craft\events\RegisterGqlMutationsEvent))
#11 /Users/ransom/Projects/Work/<redacted>/vendor/yiisoft/yii2/base/Component.php(637): yii\base\Event::trigger('craft\\services\\...', 'registerGqlMuta...', Object(craft\events\RegisterGqlMutationsEvent))
#12 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/services/Gql.php(1390): yii\base\Component->trigger('registerGqlMuta...', Object(craft\events\RegisterGqlMutationsEvent))
#13 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/services/Gql.php(396): craft\services\Gql->_registerGqlMutations()
#14 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/services/Gql.php(533): craft\services\Gql->getSchemaDef(Object(craft\models\GqlSchema), false)
#15 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/twig/Extension.php(1210): craft\services\Gql->executeQuery(Object(craft\models\GqlSchema), 'fragment LinkFi...', Array, NULL)
#16 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/81/81f59b947447ef2d5c005a9cc937268466c1650d4a2066f7cedbf074a02db09f.php(79): craft\web\twig\Extension->gqlFunction('fragment LinkFi...', Array)
#17 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(182): __TwigTemplate_0d4f14f5a5051f2bc1fa332bbd46dc3224ae4082cd4f0105ec31169a27e56b78->block_content(Array, Array)
#18 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/bd/bd237b8e4184e1e8859f27aec245d1c574fb6ed3637cba67022a0af4c7b1a7d7.php(98): Twig\Template->displayBlock('content', Array, Array)
#19 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(182): __TwigTemplate_4bf27f562b3126957a452c80d51d35bedff69db44ffcbe3c54d2a71c9cfa1df3->block_bodyHtml(Array, Array)
#20 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/22/229e514986a836cf86e1f6dfa9b7b8373511b249f6991428a108939f95a35d98.php(164): Twig\Template->displayBlock('bodyHtml', Array, Array)
#21 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(182): __TwigTemplate_83e563363af6ba080257fa3c571ed5aa678844042fc9ffec78a67e7259c76abd->block_bodyContent(Array, Array)
#22 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/18/18841e06cc3e964ce283ba32c247407403489a93a7f87c35ea816447e16b562c.php(85): Twig\Template->displayBlock('bodyContent', Array, Array)
#23 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(182): __TwigTemplate_2101bb021bf8d9866ff1d45d74f3bf1f9c83b1302efcc5b93255c28e7ff81197->block_htmlPage(Array, Array)
#24 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/ab/abd40d713a4ec1f69ae573660af3f92b84a9553d5114349cff61a4af2922ca38.php(66): Twig\Template->displayBlock('htmlPage', Array, Array)
#25 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(405): __TwigTemplate_895010e82ff53c5b008f3ce75984ec9932408cb7e7446a093281e8ae71ed8c12->doDisplay(Array, Array)
#26 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#27 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/18/18841e06cc3e964ce283ba32c247407403489a93a7f87c35ea816447e16b562c.php(48): Twig\Template->display(Array, Array)
#28 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(405): __TwigTemplate_2101bb021bf8d9866ff1d45d74f3bf1f9c83b1302efcc5b93255c28e7ff81197->doDisplay(Array, Array)
#29 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#30 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/22/229e514986a836cf86e1f6dfa9b7b8373511b249f6991428a108939f95a35d98.php(52): Twig\Template->display(Array, Array)
#31 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(405): __TwigTemplate_83e563363af6ba080257fa3c571ed5aa678844042fc9ffec78a67e7259c76abd->doDisplay(Array, Array)
#32 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#33 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/bd/bd237b8e4184e1e8859f27aec245d1c574fb6ed3637cba67022a0af4c7b1a7d7.php(50): Twig\Template->display(Array, Array)
#34 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(405): __TwigTemplate_4bf27f562b3126957a452c80d51d35bedff69db44ffcbe3c54d2a71c9cfa1df3->doDisplay(Array, Array)
#35 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#36 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/81/81f59b947447ef2d5c005a9cc937268466c1650d4a2066f7cedbf074a02db09f.php(44): Twig\Template->display(Array, Array)
#37 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(405): __TwigTemplate_0d4f14f5a5051f2bc1fa332bbd46dc3224ae4082cd4f0105ec31169a27e56b78->doDisplay(Array, Array)
#38 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#39 /Users/ransom/Projects/Work/<redacted>/storage/runtime/compiled_templates/dc/dc3b31675c967e21b7c2541b09d4e544a82b54d2a2c1baebe89e35dd32e40f94.php(40): Twig\Template->display(Array)
#40 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(405): __TwigTemplate_3f35da56241727cb11ec3329dfd8fb0c6c86eb367f88a393d00124deb880042c->doDisplay(Array, Array)
#41 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array)
#42 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Template.php(390): Twig\Template->display(Array)
#43 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/TemplateWrapper.php(45): Twig\Template->render(Array, Array)
#44 /Users/ransom/Projects/Work/<redacted>/vendor/twig/twig/src/Environment.php(318): Twig\TemplateWrapper->render(Array)
#45 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/View.php(390): Twig\Environment->render('sections/pages/...', Array)
#46 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/View.php(451): craft\web\View->renderTemplate('sections/pages/...', Array)
#47 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/Controller.php(257): craft\web\View->renderPageTemplate('sections/pages/...', Array, 'site')
#48 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/controllers/TemplatesController.php(100): craft\web\Controller->renderTemplate('sections/pages/...', Array)
#49 [internal function]: craft\controllers\TemplatesController->actionRender('sections/pages/...', Array)
#50 /Users/ransom/Projects/Work/<redacted>/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#51 /Users/ransom/Projects/Work/<redacted>/vendor/yiisoft/yii2/base/Controller.php(181): yii\base\InlineAction->runWithParams(Array)
#52 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/Controller.php(190): yii\base\Controller->runAction('render', Array)
#53 /Users/ransom/Projects/Work/<redacted>/vendor/yiisoft/yii2/base/Module.php(534): craft\web\Controller->runAction('render', Array)
#54 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/Application.php(274): yii\base\Module->runAction('templates/rende...', Array)
#55 /Users/ransom/Projects/Work/<redacted>/vendor/yiisoft/yii2/web/Application.php(104): craft\web\Application->runAction('templates/rende...', Array)
#56 /Users/ransom/Projects/Work/<redacted>/vendor/craftcms/cms/src/web/Application.php(259): yii\web\Application->handleRequest(Object(craft\web\Request))
#57 /Users/ransom/Projects/Work/<redacted>/vendor/yiisoft/yii2/base/Application.php(392): craft\web\Application->handleRequest(Object(craft\web\Request))
#58 /Users/ransom/Projects/Work/<redacted>/web/index.php(22): yii\base\Application->run()
#59 /Users/ransom/.composer/vendor/laravel/valet/server.php(219): require('/Users/ransom/P...')
#60 {main}
```

**Comment by engram-design** [comment]
@Mosnar Just pushed a fix for this if you'd like to try again!

**Comment by engram-design** [comment]
Should be fixed in [1.3.22](https://github.com/verbb/formie/releases/tag/1.3.22)

**Comment by Mosnar** [comment]
Works great, nice work!
